### PR TITLE
sched/signal: Implement SA_NODEFER

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
See the info here:
https://pubs.opengroup.org/onlinepubs/007904875/functions/sigaction.html

## Impact
New funcitonality

## Testing
Pass CI and ostest

### CODE REVIEW

File: include/signal.h [hunk 1]
```diff
@@ -249,6 +249,14 @@
                                   * (always assumed) */
 #define SA_NOCLDWAIT    (1 << 2) /* If signo=SIGCHLD, exit status of child
                                   * processes will be discarded */
+#define SA_ONSTACK      (1 << 3) /* Inicates that a registered stack_t
```
Comment (gustavonihei) [inline include/signal.h:252]:
```suggestion
#define SA_ONSTACK      (1 << 3) /* Indicates that a registered stack_t
```

File: include/signal.h [hunk 2]
```diff
@@ -249,6 +249,14 @@
...
+                                  * will be used */
+#define SA_RESTART      (1 << 4) /* flag to get restarting signals
```
Comment (gustavonihei) [inline include/signal.h:254]:
```suggestion
#define SA_RESTART      (1 << 4) /* Flag to get restarting signals
```

File: include/signal.h [hunk 3]
```diff
@@ -249,6 +249,14 @@
...
+                                  * (which were the default long ago) */
+#define SA_NODEFER      (1 << 5) /* prevents the current signal from
+                                  * being masked in the handler */
+#define SA_RESETHAND    (1 << 6) /* clears the handler when the signal
```
Comment (gustavonihei) [inline include/signal.h:258]:
```suggestion
#define SA_RESETHAND    (1 << 6) /* Clears the handler when the signal
```

File: include/signal.h [hunk 4]
```diff
@@ -249,6 +249,14 @@
                                   * (always assumed) */
 #define SA_NOCLDWAIT    (1 << 2) /* If signo=SIGCHLD, exit status of child
                                   * processes will be discarded */
+#define SA_ONSTACK      (1 << 3) /* Inicates that a registered stack_t
+                                  * will be used */
+#define SA_RESTART      (1 << 4) /* flag to get restarting signals
+                                  * (which were the default long ago) */
+#define SA_NODEFER      (1 << 5) /* prevents the current signal from
```
Comment (gustavonihei) [inline include/signal.h:256]:
```suggestion
#define SA_NODEFER      (1 << 5) /* Prevents the current signal from
```

**Review by gustavonihei — APPROVED** [review]
LGTM, but please capitalize the documentation to be consistent.

File: include/signal.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/signal.h]: Done.

File: include/signal.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/signal.h]: Done.

File: include/signal.h (hunk 3 above)
Comment (xiaoxiang781216) [reply include/signal.h]: Done.

File: include/signal.h (hunk 4 above)
Comment (xiaoxiang781216) [reply include/signal.h]: Done.

**Comment by xiaoxiang781216** [comment]
> LGTM, but please capitalize the documentation to be consistent.

Fix, thanks for reviewing.

**Pull request merged**